docs/adrs/0003-bulk-import-ui.md

```diff
@@ -0,0 +1,91 @@
+* A core system used to render the cell based on state.
+* A message bus that's used to communicate with the rest of the FE application, as well as the BE (kernel) to get job state.
+* An interpreter / manager for app specs - turns the app specs into sets of input form elements.
+* A code translator - turns the various sets of inputs into code that can be executed to run the App.
+
+The reason for the App cell being an extension to code cells is the last point above: in essence, the primary UI components of the App cell serve as a fancy code generator. User interactions with form elements generate the code that can be used to start apps and view their results. Thus, anything that an app cell does with the UI can be done programmatically by an advanced user.
+
+## Alternatives Considered
+
+* Implement the UI in the existing App Cell codebase
+* Implement the UI in a new Cell type
+* Don't use a cell, but some other interface
+
+## Decision Outcome
+
+## Consequences
+
+## Pros and Cons of the Alternatives
+
+### Implement the UI in the existing App Cell code base
+
+This option would entail adding a different mode for the current App Cell, likely based on some additional metadata feature.
+
+* `+` Reduce spread of additional, yet similar, cell types with similar code.
+* `+` Reuse existing cell control mechanisms currently in place with minimal modification.
+* `+` Reuse existing wrapping UI elements - cell title, icons, tabs, etc.
+* `-` Work done to the App cell needs to be made backward compatible with existing functionality.
+* `-` Any "breaking" work (i.e. refactor, etc.) will need to be complete before any releases are possible, making intermediate releases possibly challenging.
+* `-` Adding yet another mode for the App cell will increase the maintenance burden on an already complex code base - see especially the FSM.
+* `-` The draft UI is very complex and may turn the existing code into a deep series of nested logic.
+
+### Implement the UI in a new Cell type
+
+This option will create a new nbextension to serve a new cell type as an extension to the code cell.
+
+* `+` This would mean working with a fresh codebase without existing constraints.
+* `+` Intermediate, minimally functional changes, can be rapidly prototyped and released.
+* `-` To stick with the DRY principle, this would mean refactoring parts of the current app cell to make various components importable (message bus responses, job state management, tab menu, etc.) instead of their integration (really we might want to consider that anyway).
+* `-` Much of the structure of the mockup already functions similar to the existing App Cell and might become duplicated work with additional maintenance burden.
+* `-` This would mean updating other components that make use of Narratives - the Static Narrative interface, Search, and the Dashboard, for example, all introspect Narratives and handle cells based on their type.
```

Comment:
Can we add a bit on how it may require changing importer app specs?

Reply:
I added a block about possibly changing app specs and how they function.